File: library/src/main/java/com/getstream/sdk/chat/storage/converter/QuerySortConverter.java
package com.getstream.sdk.chat.storage.converter;

import androidx.room.TypeConverter;

import com.getstream.sdk.chat.enums.QuerySort;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;

public class QuerySortConverter {
    static Gson gson = new Gson();

    @TypeConverter
    public static QuerySort stringToObject(String data) {
        if (data == null) {
            return new QuerySort();
        }

        Type listType = new TypeToken<QuerySort>() {
        }.getType();

        return gson.fromJson(data, listType);
    }

    @TypeConverter
    public static String objectToString(QuerySort someObjects) {
        return gson.toJson(someObjects);
    }
}

File: app/src/main/java/com/getstream/sdk/chat/livedata/converter/QuerySortConverter.kt
package com.getstream.sdk.chat.livedata.converter

import androidx.room.TypeConverter
import com.getstream.sdk.chat.livedata.gson
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import io.getstream.chat.android.client.api.models.QuerySort

object QuerySortConverter {
    @TypeConverter
    @JvmStatic
    fun stringToObject(data: String?): QuerySort {
        if (data == null) {
            return QuerySort()
        }
        val listType = object : TypeToken<QuerySort?>() {}.type
        return gson.fromJson(data, listType)
    }

    @TypeConverter
    @JvmStatic
    fun objectToString(someObjects: QuerySort?): String {
        return gson.toJson(someObjects)
    }
}
